### PR TITLE
API-1255 : Get back options labels in the product call

### PR DESCRIPTION
**Pull request by tomglvng**
Adds query parameter (no BC break and perf altered only if wanted) called “with_attribute_options”.

### CODE REVIEW

**Comment by pierrejolly** [comment]
I think you miss to update the endpoint to get a single product

File: content/swagger/parameters.yaml [hunk 1]
```diff
@@ -85,3 +85,10 @@ with_count:
   default: false
   type: boolean
   required: false
+with_attribute_options:
```
Comment (NolwennP) [inline content/swagger/parameters.yaml:88]: The naming of this query parameter does not seem really clear as we already return by default the attribute options. But the code, not the label. So I would suggest:
```suggestion
with_attribute_option_labels:
```
@meganelepalud-akeneo What do you think?

**Review by NolwennP — COMMENTED** [review]
You also need to specify in the Swagger the changes it implies in the product format returned by the API in the case this parameter is used.

File: content/swagger/resources/products/routes/products.yaml [hunk 2]
```diff
@@ -36,6 +36,7 @@ get:
     - $ref: '#/parameters/search_after'
     - $ref: '#/parameters/limit'
     - $ref: '#/parameters/with_count'
+    - $ref: '#/parameters/with_attribute_options'
```
Comment (NolwennP) [inline content/swagger/resources/products/routes/products.yaml:39]:
```suggestion
    - $ref: '#/parameters/with_attribute_option_labels'
```

File: content/swagger/resources/products/routes/products_code.yaml [hunk 3]
```diff
@@ -12,6 +12,7 @@ get:
   description: This endpoint allows you to get the information about a given product. In the Entreprise Edition, since the v2.0, permissions based on your user groups are applied to the product you request.
   parameters:
     - $ref: '#/parameters/code'
+    - $ref: '#/parameters/with_attribute_options'
```
Comment (NolwennP) [inline content/swagger/resources/products/routes/products_code.yaml:15]:
```suggestion
    - $ref: '#/parameters/with_attribute_option_labels'
```

File: content/swagger/parameters.yaml [hunk 4]
```diff
@@ -85,3 +85,10 @@ with_count:
   default: false
   type: boolean
   required: false
+with_attribute_option_labels:
+  name: with_attribute_options
```
Comment (meganelepalud-akeneo) [inline content/swagger/parameters.yaml:89]:
```suggestion
  name: with_attribute_options_labels
```

**Comment by fitn** [comment]
`with_attribute_options` is the expected parameter. We may add more information later if needed with the same parameter (like the sort_order for instance).

**Review by meganelepalud-akeneo — APPROVED** [review]

**Pull request merged**